audiostream/inc/ni/media/audio/ifstream_support.h

```diff
@@ -24,16 +24,16 @@
 
 #include <ni/media/audio/ifstream_info.h>
 
+#include <filesystem>
 #include <map>
-#include <string>
 
 namespace audio
 {
 
 // a map of extension => container type
-using ifstream_container_map = std::map<std::string, ifstream_info::container_type>;
+using ifstream_container_map = std::map<std::filesystem::path, ifstream_info::container_type>;
```

Comment:
Here I wasn't sure if this might be overkill but motivation was that [extension()](https://en.cppreference.com/w/cpp/filesystem/path/extension) also returns a `path`.